pandas/core/common.py

```diff
@@ -220,6 +223,17 @@ def asarray_tuplesafe(values, dtype=None):
     if isinstance(values, list) and dtype in [np.object_, object]:
         return construct_1d_object_array_from_listlike(values)
 
+    if isinstance(values, list) and hasattr(values, "__array__"):
+        # avoid converting extension array to numpy array
+        return values
+
+    if (
+        isinstance(values, list)
```

Comment:
really avoid doing things like this, what are you trying to do here?

Reply:
@jreback 
Okay. i changed code. i want to infer indexer are list of strings representing datetime or periods or timedelta.